src/webkit/wkWorkers.ts

```diff
@@ -47,17 +47,15 @@ export class WKWorkers {
         worker._createExecutionContext(new WKExecutionContext(workerSession, undefined));
         this._page._addWorker(event.workerId, worker);
         workerSession.on('Console.messageAdded', event => this._onConsoleMessage(worker, event));
-        try {
-          Promise.all([
-            workerSession.send('Runtime.enable'),
-            workerSession.send('Console.enable'),
-            session.send('Worker.initialized', { workerId: event.workerId }).catch(e => {
-              this._page._removeWorker(event.workerId);
-            })
-          ]);
-        } catch (e) {
+        Promise.all([
+          workerSession.send('Runtime.enable'),
+          workerSession.send('Console.enable'),
+          session.send('Worker.initialized', { workerId: event.workerId }).catch(e => {
+            this._page._removeWorker(event.workerId);
+          })
+        ]).catch(e => {
           // Worker can go as we are initializing it.
-        }
+        });
```

Comment:
Should we call `page._removeWorker` here, and not in line 54?

Reply:
Makes sense. Done.